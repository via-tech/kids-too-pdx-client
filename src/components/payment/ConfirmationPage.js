import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function ConfirmationPage({ subFee, confirmation }) {
  if(confirmation) return (
    <section>
      <p>Payment successful! Subscription will automatically renew each year. If on trial period, a ${subFee} subscription fee will automatically be charged after trial ends. Email reminders will be sent out accordingly. To manage subscription preferences, please visit the dashboard. Thank you!</p>

      <p><strong>Please verify your email address with the link sent to the provided email.</strong></p>

      <p>
        Continue to
        <Link to="/">Home</Link> page
      </p>
    </section>
  );

  return <p>
    Confirmation page only upon payment!
    Go <Link to="/">Home</Link>
  </p>;
}

ConfirmationPage.propTypes = {
  subFee: PropTypes.string.isRequired,
  confirmation: PropTypes.bool.isRequired
};

export default ConfirmationPage;
