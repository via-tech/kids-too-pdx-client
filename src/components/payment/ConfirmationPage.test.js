import React from 'react';
import renderer from 'react-test-renderer';
import ConfirmationPage from './ConfirmationPage';
import { MemoryRouter } from 'react-router-dom';

describe('ConfirmationPage', () => {
  it('snapshots confirmation page', () => {
    const props = {
      subFee: '500',
      confirmation: true
    };

    const tree = renderer.create(
      <MemoryRouter>
        <ConfirmationPage {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots confirmation page without confirmation', () => {
    const props = {
      subFee: '500',
      confirmation: false
    };

    const tree = renderer.create(
      <MemoryRouter>
        <ConfirmationPage {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
