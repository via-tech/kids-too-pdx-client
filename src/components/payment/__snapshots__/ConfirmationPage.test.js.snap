// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPage snapshots confirmation page 1`] = `
<section>
  <p>
    Payment successful! Subscription will automatically renew each year. If on trial period, a $
    500
     subscription fee will automatically be charged after trial ends. Email reminders will be sent out accordingly. To manage subscription preferences, please visit the dashboard. Thank you!
  </p>
  <p>
    <strong>
      Please verify your email address with the link sent to the provided email.
    </strong>
  </p>
  <p>
    Continue to
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
     page
  </p>
</section>
`;

exports[`ConfirmationPage snapshots confirmation page without confirmation 1`] = `
<p>
  Confirmation page only upon payment! Go 
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</p>
`;
